pallets/liquidity-pools/src/message.rs

```diff
@@ -406,6 +528,28 @@ pub enum Message {
 	},
 }
 
+impl Message {
+	/// Compose this message with a new one
+	pub fn pack(&self, other: Self) -> Result<Self, DispatchError> {
+		Ok(match self.clone() {
+			Message::Batch(content) => {
+				let mut content = content.clone();
+				content.try_add(other)?;
+				Message::Batch(content)
+			}
+			this => Message::Batch(BatchMessages::try_from(vec![this.clone(), other])?),
+		})
+	}
+
+	/// Decompose the message into a list of messages
+	pub fn unpack(&self) -> Vec<Self> {
+		match self {
+			Message::Batch(content) => content.clone().into_iter().collect(),
```

Comment:
I think a `BoundedVec` does not need to be collected. Can use `Into` or something

Reply:
Good point!

Reply:
Ah, I remember why I did it so. I need to iterate to remove the `NoBatchMessage` wrapper of each message. The `into_iter()` does not belong to `BoundedVec`, it belongs to `BatchMessages` which automatically unwraps the `NoBatchmessage` into normal `Message`.